### ISSUE TITLE
Got `Error: This socket is closed` after refreshing the browser

### ISSUE THREAD

**Issue by kittaakos** [eclipse-theia/theia#2113]
This happened when I have killed and restarted the backend process and then refreshed the browser window. I was unable to reproduce this defect consistently.
```
[2018-06-14T11:20:24.981Z] ERROR: Theia/43591 on Akoss-MacBook-Pro.local: Uncaught Exception:  [ 'Error: This socket is closed' ] (logger=root)

[2018-06-14T11:20:24.992Z] ERROR: Theia/43591 on Akoss-MacBook-Pro.local: (logger=root)
    Error: This socket is closed
        at Socket._writeGeneric (net.js:727:18)
        at Socket._write (net.js:787:8)
        at doWrite (_stream_writable.js:387:12)
        at writeOrBuffer (_stream_writable.js:373:5)
        at Socket.Writable.write (_stream_writable.js:290:11)
        at Socket.write (net.js:705:40)
        at Sender.sendFrame (/Users/akos.kitta/git/theia/packages/core/node_modules/ws/lib/Sender.js:390:20)
        at Sender.send (/Users/akos.kitta/git/theia/packages/core/node_modules/ws/lib/Sender.js:312:12)
        at WebSocket.send (/Users/akos.kitta/git/theia/packages/core/node_modules/ws/lib/WebSocket.js:377:18)
        at WebSocketChannel.doSend (/Users/akos.kitta/git/theia/packages/core/lib/node/messaging/messaging-contribution.js:189:97) []
```

**Comment by nchristensen** [comment]
This similarly happens to me on my Linux machine when I kill and restart the backend process, refresh the browser, and then attempt to open a file. Files fail to open until I restart the computer. 

```
[2018-06-19T23:58:01.035Z] ERROR: Theia/30675 on test-Precision-Tower-7910: Uncaught Exception:  [ 'Error: socket hang up' ] (logger=root)

[2018-06-19T23:58:01.036Z] ERROR: Theia/30675 on test-Precision-Tower-7910: (logger=root)
    Error: socket hang up
        at createHangUpError (_http_client.js:331:15)
        at TLSSocket.socketCloseListener (_http_client.js:363:23)
        at emitOne (events.js:121:20)
        at TLSSocket.emit (events.js:211:7)
        at _handle.close (net.js:557:12)
        at TCP.done [as _onclose] (_tls_wrap.js:356:7) []
```

**Comment by JonasHelming** [comment]
I have not experiences this, although I am on Linux and I do the steps described above quite often. Maybe it was fixed with #2572
I suggest to close this and reopne if somebody can reproduce it again. @kittaakos Are you fine with this?
